ngraph/frontend/generic/include/frontend_manager/frontend_manager.hpp

```diff
@@ -0,0 +1,454 @@
+
+            /// \brief Loads an input model by already loaded memory buffer
+            /// Memory structure is frontend-defined and is not specified in generic API
+            virtual InputModel::Ptr load_from_memory(const void* model) const;
+
+            /// \brief Loads an input model from set of memory buffers
+            /// Memory structure is frontend-defined and is not specified in generic API
+            virtual InputModel::Ptr
+                load_from_memory_fragments(const std::vector<const void*>& modelParts) const;
+
+            /// \brief Loads an input model by input stream representing main model file
+            virtual InputModel::Ptr load_from_stream(std::istream& path) const;
+
+            /// \brief Loads an input model by input streams representing all model files
+            virtual InputModel::Ptr
+                load_from_streams(const std::vector<std::istream*>& paths) const;
+
+            /// \brief Completely convert and normalize entire function, throws if it is not
+            /// possible
+            virtual std::shared_ptr<ngraph::Function> convert(InputModel::Ptr model) const;
+
+            /// \brief Completely convert the remaining, not converted part of a function.
+            virtual std::shared_ptr<ngraph::Function>
+                convert(std::shared_ptr<ngraph::Function> partiallyConverted) const;
+
+            /// \brief Convert only those parts of the model that can be converted leaving others
+            /// as-is. Converted parts are not normalized by additional transformations; normalize
+            /// function or another form of convert function should be called to finalize the
+            /// conversion process.
+            virtual std::shared_ptr<ngraph::Function>
+                convert_partially(InputModel::Ptr model) const;
+
+            /// \brief Convert operations with one-to-one mapping with decoding nodes.
+            /// Each decoding node is an nGraph node representing a single FW operation node with
+            /// all attributes represented in FW-independent way.
+            virtual std::shared_ptr<ngraph::Function> decode(InputModel::Ptr model) const;
+
+            /// \brief Runs normalization passes on function that was loaded with partial conversion
+            virtual void normalize(std::shared_ptr<ngraph::Function> function) const;
+        };
+
+        /// Capabilities for requested FrontEnd
+        /// In general, frontend implementation may be divided into several libraries by capability
+        /// level It will allow faster load of frontend when only limited usage is expected by
+        /// client application as well as binary size can be minimized by removing not needed parts
+        /// from application's package
+        namespace FrontEndCapabilities
+        {
+            /// \brief Just reading and conversion, w/o any modifications; intended to be used in
+            /// Reader
+            static const int FEC_DEFAULT = 0;
```

Comment:
Can it be enum?

Reply:
It was an enum in initial commit, but it was found that operating with enum like with flags is very inconvenient. We shall be able to give possibility to ```loadByFramework("pdpd", FEC_CUT | FEC_WILDCARDS)```. And plugin's code shall be able to check enabled flags with ```if (fec & FEC_CUT)```. Having "enum class" will require defining custom operators "|" and "&" to do such operations. So, I checked "istream" code, which also have flags like "std::ios::binary", and this is number of constants